### ISSUE TITLE
Fix Request: false positive for AvoidIdentityCacheKeys when used SimpleKey

### ISSUE THREAD

**Issue by hem-verma** [jborgers/PMD-jPinpoint-rules#149]
Getting AvoidIdentityCacheKeys when used SimpleKey as a recommended approach to fix it. SimpleKey uses the hashcode and equals to generate the key.

```
@Override
    public Object generate(Object target, Method method, Object... params) {
        return new SimpleKey(target.getClass().getName(), method.getName(), params);
    }
```
